src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1047,43 +1104,34 @@ export default class ComparisonStore {
         },
     });
 
-    readonly overallPatientSurvivals = remoteData<PatientSurvival[]>(
-        {
-            await: () => [
-                this.survivalClinicalDataGroupByUniquePatientKey,
-                this.activePatientKeysNotOverlapRemoved,
-            ],
-            invoke: async () => {
-                return getPatientSurvivals(
-                    this.survivalClinicalDataGroupByUniquePatientKey.result,
-                    this.activePatientKeysNotOverlapRemoved.result!,
-                    'OS_STATUS',
-                    'OS_MONTHS',
-                    s => s === 'DECEASED'
-                );
-            },
-        },
-        []
-    );
-
-    readonly diseaseFreePatientSurvivals = remoteData<PatientSurvival[]>(
-        {
-            await: () => [
-                this.survivalClinicalDataGroupByUniquePatientKey,
-                this.activePatientKeysNotOverlapRemoved,
-            ],
-            invoke: async () => {
-                return getPatientSurvivals(
-                    this.survivalClinicalDataGroupByUniquePatientKey.result,
-                    this.activePatientKeysNotOverlapRemoved.result!,
-                    'DFS_STATUS',
-                    'DFS_MONTHS',
-                    s => s === 'Recurred/Progressed' || s === 'Recurred'
-                );
-            },
+    readonly patientSurvivals = remoteData<{
+        [prefix: string]: PatientSurvival[];
+    }>({
+        await: () => [
+            this.survivalClinicalDataGroupByUniquePatientKey,
+            this.activePatientKeysNotOverlapRemoved,
+            this.survivalClinicalAttributesPrefix,
+        ],
+        invoke: () => {
+            return Promise.resolve(
+                _.reduce(
+                    this.survivalClinicalAttributesPrefix.result!,
+                    (result, key) => {
```

Comment:
using "result" here is a little confusing because we are processing the result property of the promise.   we usually use "acc" (accumulator) as the name of this argument.